internal/hcs/system.go

```diff
@@ -25,102 +25,92 @@ type System struct {
 	callbackNumber uintptr
 }
 
-// createContainerAdditionalJSON is read from the environment at initialisation
+// createComputeSystemAdditionalJSON is read from the environment at initialisation
 // time. It allows an environment variable to define additional JSON which
-// is merged in the CreateContainer call to HCS.
-var createContainerAdditionalJSON string
+// is merged in the CreateComputeSystem call to HCS.
+var createComputeSystemAdditionalJSON string
 
 func init() {
-	createContainerAdditionalJSON = os.Getenv("HCSSHIM_CREATECONTAINER_ADDITIONALJSON")
+	createComputeSystemAdditionalJSON = os.Getenv("HCSSHIM_CREATECOMPUTESYSTEM_ADDITIONALJSON")
 }
 
-// CreateContainer creates a new container with the given configuration but does not start it.
-func CreateContainer(id string, c *schema1.ContainerConfig) (*System, error) {
-	return createContainerWithJSON(id, c, "")
-}
-
-// CreateContainerWithJSON creates a new container with the given configuration but does not start it.
-// It is identical to CreateContainer except that optional additional JSON can be merged before passing to HCS.
-func CreateContainerWithJSON(id string, c *schema1.ContainerConfig, additionalJSON string) (*System, error) {
-	return createContainerWithJSON(id, c, additionalJSON)
-}
-
-func createContainerWithJSON(id string, c *schema1.ContainerConfig, additionalJSON string) (*System, error) {
-	operation := "CreateContainer"
+// CreateComputeSystem creates a new compute system with the given configuration but does not start it.
+func CreateComputeSystem(id string, hcsDocumentInterface interface{}, additionalJSON string) (*System, error) {
```

Comment:
I think the merging and additional parameter should be removed from this level, and the handling should be moved up to the existing container object. If we want to keep the environment variable model for extending the JSON (and I'm not sure we do; runhcs can handle this is other ways), then we probably need separate env vars for utility VM vs container.

Reply:
We can tweak this later, so it doesn't need to be changed for this PR. The breaking change question needs to be answered though.

Reply:
Yes, I agree with this. Can tweak later, certainly.